rasa_core/policies/form_policy.py

```diff
@@ -6,48 +6,130 @@
 import logging
 import typing
 
-from typing import Any, List, Text
+from typing import Any, List, Optional, Dict
 
 from rasa_core.constants import FORM_SCORE
-from rasa_core.policies import Policy
+from rasa_core.policies.memoization import MemoizationPolicy
+from rasa_core.events import ActionExecuted, ActionExecutionRejected
 
-from rasa_core.actions.action import ACTION_LISTEN_NAME
+from rasa_core.domain import PREV_PREFIX, ACTIVE_FORM_PREFIX
+from rasa_core.actions.action import (ACTION_LISTEN_NAME,
+                                      ACTION_NO_FORM_VALIDATION_NAME)
 
 if typing.TYPE_CHECKING:
     from rasa_core.domain import Domain
     from rasa_core.trackers import DialogueStateTracker
+    from rasa_core.featurizers import TrackerFeaturizer
 
 logger = logging.getLogger(__name__)
 
 
-class FormPolicy(Policy):
+class FormPolicy(MemoizationPolicy):
     """Policy which handles deterministic prediction of Forms"""
 
-    @staticmethod
-    def _standard_featurizer():
-        return None
+    ENABLE_FEATURE_STRING_COMPRESSION = True
+
+    def __init__(self,
+                 featurizer=None,  # type: Optional[TrackerFeaturizer]
+                 lookup=None  # type: Optional[Dict]
+                 ):
+        # type: (...) -> None
+
+        super(FormPolicy, self).__init__(featurizer=featurizer,
+                                         max_history=1,
+                                         lookup=lookup)
 
     def train(self,
               training_trackers,  # type: List[DialogueStateTracker]
               domain,  # type: Domain
               **kwargs  # type: Any
               ):
         # type: (...) -> None
-        """Does nothing. This policy is deterministic."""
+        """Finds intents that shouldn't be validated.
+            Assumes that stories contain only unhappy paths"""
+
+        self.lookup = {}
 
-        pass
+        trackers_as_states, _ = self.featurizer.training_states_and_actions(
```

Comment:
just for me to clarify, `max_history == 1` so this will contain only one state right?

Reply:
yes, we do not try to learn what to predict, but rather when not to predict